functions/Get-DbaBackupHistory.ps1

```diff
@@ -237,15 +257,21 @@ function Get-DbaBackupHistory {
 				$DeviceTypeFilterRight = "IN ('" + ($DeviceTypeFilter -Join "','") + "')"
 			}
 			
+			foreach($b in $BackupTypeFilter) {
+				$BackupTypeFilterRight = "IN ('" + ($BackupTypeFilter -Join "','") + "')"
+			}
+			
 			if ($last) {
 				
-				foreach ($db in $databases.Name) {
+				foreach ($db in $databases) {
 					
 					#Get the full and build upwards
 					$allbackups = @()
-					$allbackups += $Fulldb = Get-DbaBackupHistory -SqlInstance $server -Database $db -LastFull -raw:$Raw -DeviceType $DeviceType
-					$DiffDB = Get-DbaBackupHistory -SqlInstance $server -Database $db -LastDiff -raw:$Raw -DeviceType $DeviceType
+					$allbackups += $Fulldb = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -LastFull -raw:$Raw -DeviceType $DeviceType
+					$DiffDB = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -LastDiff -raw:$Raw -DeviceType $DeviceType
 					if ($DiffDb.LastLsn -gt $Fulldb.LastLsn -and  $DiffDb.DatabaseBackupLSN -eq $Fulldb.CheckPointLSN ) {
+						$Allbackups += $DiffDB = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -LastDiff -raw:$Raw -DeviceType $DeviceType
```

Comment:
Isn't this repeating the call on L271? Just $Allbackups += $DiffDB could be used?